### PR TITLE
[clang-analyze] Fix assorted clang-analyze warnings

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Fix various warnings from a clang-analyze report on macos build.

## Motivation and context
Fix some warnings from clang-analyze. This is only some of the low hanging warnings

## How has this been tested?
build osx

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp [hunk 1]
```diff
@@ -184,7 +184,7 @@ void CDVDAudioCodecFFmpeg::GetData(DVDAudioFrame &frame)
 {
   frame.nb_frames = 0;
 
-  uint8_t* data[16];
+  uint8_t* data[16]{};
```
Comment (lrusak) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:187]: This seems odd as it's used by the next line.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ void CDVDVideoPPFFmpeg::Process(VideoPicture* pPicture)
   }
 
   uint8_t* srcPlanes[YuvImage::MAX_PLANES], *dstPlanes[YuvImage::MAX_PLANES];
-  int srcStrides[YuvImage::MAX_PLANES];
+  int srcStrides[YuvImage::MAX_PLANES]{};
```
Comment (lrusak) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp:99]: Same thing here. Seems odd but I guess default init doesn't hurt.

File: xbmc/guilib/GUIFontTTFGL.cpp [hunk 3]
```diff
@@ -139,18 +139,6 @@ void CGUIFontTTFGL::LastEnd()
 
     // Deal with vertices that had to use software clipping
     std::vector<SVertex> vecVertices(6 * (m_vertex.size() / 4));
-    SVertex *vertices = &vecVertices[0];
```
Comment (lrusak) [inline xbmc/guilib/GUIFontTTFGL.cpp:142]: This seems wrong. This block adds the vertices needed to vecVertives which is used in the gl calls below. There is some pointer addition happening which may be why it thinks the variable is unused?

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:187]: You could use `std::array`

File: xbmc/guilib/GUIFontTTFGL.cpp (hunk 3 above)
Comment (fuzzard) [reply xbmc/guilib/GUIFontTTFGL.cpp:142]: Yeah, theres some pointer arithmetic, but in the end vertices isnt used for anything in this HAS_GL codepath.
In the GLES equivalent, its used in the following https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L199-L201

Whereas the GL path does https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L149-L154

Comment (lrusak) [reply xbmc/guilib/GUIFontTTFGL.cpp:142]: The gl method uses a GL_ARRAY_BUFFER

https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L147

Comment (fuzzard) [reply xbmc/guilib/GUIFontTTFGL.cpp:142]: ah, ok, i see. pointer to the pointer to the reference. Ill drop this then. drum it up to a false positive

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp (hunk 2 above)
Comment (fuzzard) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp:99]: for the sake of some brackets to remove a warning from a static analysis tool, id just put it in personally.

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp (hunk 1 above)
Comment (fuzzard) [reply xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:187]: RE std:array, not really looking to change behaviour at this point.
Same reasoning as the other one, a couple brackets removes a static analysis warning.

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp [hunk 4]
```diff
@@ -134,7 +133,7 @@ bool CDVDAudioCodecPassthrough::AddData(const DemuxPacket &packet)
       return true;
 
     m_dataSize = m_bufferSize;
-    used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
+    int used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
```
Comment (kambala-decapitator) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp:136]:
```suggestion
    auto used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
```

**Review by kambala-decapitator — COMMENTED** [review]
isn't it possible to replace `= 0` with `{}`?

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp (hunk 4 above)
Comment (fuzzard) [reply xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp:136]: there was a discussion around this sort of change not too long ago. I think i am on the side of ints a basic type, why go auto when its longer than int to type

**Comment by fuzzard** [comment]
Any complaints with this as is? 
If not, i'd like to merge it this weekend

File: xbmc/dialogs/GUIDialogMediaFilter.cpp [hunk 5]
```diff
@@ -474,7 +474,9 @@ void CGUIDialogMediaFilter::InitializeSettings()
 
       if (filter.settingType == SettingType::Integer)
       {
-        int min, interval, max;
+        int min = 0;
```
Comment (howie-f) [inline xbmc/dialogs/GUIDialogMediaFilter.cpp:477]: no complaints, just a remark https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#54-initialization => `int min{};`
would be in line with the arrays then, but honestly my personal preference is your version

**Review by howie-f — APPROVED** [review]

**Pull request merged**